### PR TITLE
document --inject-checksums

### PR DESCRIPTION
**Pull request by boegel**
(cfr. https://github.com/easybuilders/easybuild-framework/pull/2286)

preview at http://boegel-eb.readthedocs.io/en/inject_checksums/Writing_easyconfig_files.html#inject-checksums

### CODE REVIEW

File: docs/Writing_easyconfig_files.rst [hunk 1]
```diff
@@ -180,7 +186,162 @@ of the form ``('<checksum value>', '<checksum type>')``.
...
+.. _inject_checksums:
+
+Adding or replacing checksums using ``--inject-checksums``
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Using the ``--inject-checksums`` command line option, you can let EasyBuild add or update checksums
+in one or more easyconfig files (which is significantly more convenient than doing it manually).
+
+With ``--inject-checksums``, checksums are injected for all sources and patches (if any),
+as well as for all sources & patches of every extension listed in ``exst_list`` (if any).
```
Comment (ocaisa) [inline docs/Writing_easyconfig_files.rst]: `exts_list` I suspect

File: docs/Writing_easyconfig_files.rst [hunk 2]
```diff
@@ -180,7 +186,162 @@ of the form ``('<checksum value>', '<checksum type>')``.
...
+          which also applies to the order of specified easyconfig parameters.
+
+.. _inject_checksums_replacing:
+
+Replacing existing checksums
+****************************
+
+When one or more checksums are already specified, EasyBuild requires the use of ``--force`` together
+with ``--inject-checksums`` to replace those checksums. A clear warning will be printed to notify
+that exsiting checksums will be replaced.
```
Comment (ocaisa) [inline docs/Writing_easyconfig_files.rst]: `existing`

File: docs/Writing_easyconfig_files.rst [hunk 3]
```diff
@@ -180,7 +186,162 @@ of the form ``('<checksum value>', '<checksum type>')``.
...
+
+    WARNING: Found existing checksums in bzip2-1.0.6.eb, overwriting them (due to use of --force)...
+
+    == backup of easyconfig file saved to /example/bzip2-1.0.6.eb.bak_20170824203850...
+    == injecting sha256 checksums for sources & patches in bzip2-1.0.6.eb...
+    ...
+
+.. note::
+    Any existing checksums are *blindly* replaced when ``--inject-checksums --force`` is used:
+    the existing checksumsthey are *not verified* to be correct as during normal use of EasyBuild
```
Comment (ocaisa) [inline docs/Writing_easyconfig_files.rst]: `checksumsthey` -> `checksums`

File: docs/Writing_easyconfig_files.rst [hunk 4]
```diff
@@ -180,7 +186,162 @@ of the form ``('<checksum value>', '<checksum type>')``.
...
+    (since that would kind of defeat the purpose of ``--inject-checksums``).
+
+    In addition, it also doesn't matter whether or not checksums are available for all sources & patches:
+    with ``--inject-checksums``, checksums will be added for *all* sources and patches,
+    including for extensions listed in ``exst_list`` (if any).
```
Comment (ocaisa) [inline docs/Writing_easyconfig_files.rst]: `exts_list`...now I'm starting to doubt myself

Comment (boegel) [reply docs/Writing_easyconfig_files.rst]: No, you're right, I was just making that typo consistently... ;-)

```
$ eb -a | grep exts_list
exts_list           List with extensions added to the base installation [default: []]

**Comment by ocaisa** [comment]
Ok, lgtm

**Comment by boegel** [comment]
Merged a little prematurely since https://github.com/easybuilders/easybuild-framework/pull/2286 is not merged yet, but it's not a big issue, just as long @wpoely86 merges https://github.com/easybuilders/easybuild-framework/pull/2286 soon ;)

**Pull request merged**